controllers/metal3machine_controller.go

```diff
@@ -185,6 +185,7 @@ func patchMetal3Machine(ctx context.Context, patchHelper *patch.Helper, metal3Ma
 	conditions.SetSummary(metal3Machine,
 		conditions.WithConditions(
 			infrav1.AssociateBMHCondition,
+			infrav1.Metal3DataReadyCondition,
```

Comment:
setting the condition here, but actual patching in conditions seems to be happening later in line 194-202